keras_preprocessing/image.py

```diff
@@ -527,7 +532,21 @@ def load_img(path, grayscale=False, color_mode='rgb', target_size=None,
                         interpolation,
                         ", ".join(_PIL_INTERPOLATION_METHODS.keys())))
             resample = _PIL_INTERPOLATION_METHODS[interpolation]
-            img = img.resize(width_height_tuple, resample)
+            if not keep_aspect_ratio:
+                img = img.resize(width_height_tuple, resample)
+            else:
+                img.thumbnail(width_height_tuple, resample)
```

Comment:
If size is smaller than width_height_tuple, then thumbnail will contain small image with edges on the right and at the bottom. I'm sure it is possible to resize an image to minimal height/width keeping aspect ratio.